### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by yurypat**

### CODE REVIEW

File: src/Mews.Fiscalization.Greece/Model/Types/TaxIdentifier.cs [hunk 1]
```diff
@@ -1,20 +0,0 @@
-﻿using System;
-
-namespace Mews.Fiscalization.Greece.Model.Types
-{
-    public class TaxIdentifier : StringIdentifier
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/Types/TaxIdentifier.cs:5]: I would keep this and implement `IsValid`

File: src/Mews.Fiscalization.Greece/Model/VatExemption.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public enum VatExemption
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/VatExemption.cs:7]: `VatExemptionType`?

File: src/Mews.Fiscalization.Greece/Model/InvoiceSummary.cs [hunk 3]
```diff
@@ -5,13 +5,14 @@
 
 namespace Mews.Fiscalization.Greece.Model
 {
-    public class InvoiceRecordSummary
+    public class InvoiceSummary
     {
-        public InvoiceRecordSummary(Amount totalNetValue, Amount totalVatAmount, Amount totalGrossValue, IEnumerable<InvoiceRecordIncomeClassification> invoiceRecordIncomeClassification)
+        public InvoiceSummary(Amount totalNetValue, Amount totalVatAmount, Amount totalGrossValue, IEnumerable<ItemIncomeClassification> invoiceRecordIncomeClassification, Amount totalOtherTaxesAmount = null)
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/InvoiceSummary.cs:10]:
```suggestion
        public InvoiceSummary(Amount net, Amount vat, Amount gross, IEnumerable<ItemIncomeClassification> invoiceIncomeClassification, Amount otherTaxes = null)
```

File: src/Mews.Fiscalization.Greece/Model/ItemIncomeClassification.cs [hunk 4]
```diff
@@ -3,9 +3,9 @@
 
 namespace Mews.Fiscalization.Greece.Model
 {
-    public class InvoiceRecordIncomeClassification
+    public class ItemIncomeClassification
     {
-        public InvoiceRecordIncomeClassification(ClassificationType classificationType, ClassificationCategory classificationCategory, Amount amount)
+        public ItemIncomeClassification(ClassificationType classificationType, ClassificationCategory classificationCategory, Amount amount)
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/ItemIncomeClassification.cs:8]: Maybe just `IncomeClassification`?

File: src/Mews.Fiscalization.Greece/Model/RevenueItem.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        public TaxType TaxType { get; }
+
+        public Amount VatValue { get; }
+
+        public VatExemption? VatExemption { get; }
+
+        public CityTax CityTax { get; }
+
+        public IEnumerable<ItemIncomeClassification> InvoiceRecordIncomeClassification { get; }
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Model/RevenueItem.cs:47]:
```suggestion
        public IEnumerable<ItemIncomeClassification> IncomeClassifications { get; }
```

File: src/Mews.Fiscalization.Greece/Mapper/InvoiceDocumentMapper.cs [hunk 6]
```diff
@@ -63,161 +62,268 @@ private InvoiceParty GetInvoiceParty(InvoiceRecordParty invoiceRecordParty)
...
+                case VatExemption.WithoutVatArticle40:
+                    return Dto.Xsd.VatExemptionCategory.WithoutVatArticle40;
+                case VatExemption.WithoutVatArticle41:
+                    return Dto.Xsd.VatExemptionCategory.WithoutVatArticle41;
+                case VatExemption.WithoutVatArticle47:
+                    return Dto.Xsd.VatExemptionCategory.WithoutVatArticle47;
+                case VatExemption.WithoutVatArticle5:
+                    return Dto.Xsd.VatExemptionCategory.WithoutVatArticle5;
+                default:
+                    throw new ArgumentException($"Cannot map VatExemption {vatExemption} to Dto.Xsd.VatExemptionCategory.");
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Mapper/InvoiceDocumentMapper.cs:306]:
```suggestion
                    throw new ArgumentException($"Cannot map VatExemption {vatExemption} to {nameof(Dto.Xsd.VatExemptionCategory)}.");
```

File: src/Mews.Fiscalization.Greece/Mapper/InvoiceDocumentMapper.cs [hunk 7]
```diff
@@ -63,161 +62,268 @@ private InvoiceParty GetInvoiceParty(InvoiceRecordParty invoiceRecordParty)
...
+                case CityTaxType.Hotels3Stars:
+                    return Dto.Xsd.OtherTaxCategory.Hotels3Stars;
+                case CityTaxType.Hotels4Stars:
+                    return Dto.Xsd.OtherTaxCategory.Hotels4Stars;
+                case CityTaxType.Hotels5Stars:
+                    return Dto.Xsd.OtherTaxCategory.Hotels5Stars;
+                case CityTaxType.RoomsOrApartments:
+                    return Dto.Xsd.OtherTaxCategory.RoomsOrApartments;
+                default:
+                    throw new ArgumentException($"Cannot map CityTaxType {cityTaxType} to OtherTaxCategory.");
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalization.Greece/Mapper/InvoiceDocumentMapper.cs:325]: same here

File: src/Mews.Fiscalization.Greece/Model/InvoiceDocument.cs [hunk 8]
```diff
@@ -6,7 +6,7 @@ namespace Mews.Fiscalization.Greece.Model
 {
     public class InvoiceDocument
     {
-        public InvoiceDocument(IEnumerable<InvoiceRecord> invoiceRecords)
+        public InvoiceDocument(IEnumerable<Invoice> invoiceRecords)
```
[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/InvoiceDocument.cs:9]: Will change, thanks! Missed to rename this during models renaming

File: src/Mews.Fiscalization.Greece/Model/ItemIncomeClassification.cs (hunk 4 above)
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/ItemIncomeClassification.cs:8]: Pavel has suggested this name, https://github.com/MewsSystems/fiscalization-greece/pull/8#discussion_r487027807. But we can change this, WDYT?

File: src/Mews.Fiscalization.Greece/Model/InvoiceDocument.cs [hunk 9]
```diff
@@ -16,6 +16,6 @@ public InvoiceDocument(IEnumerable<InvoiceRecord> invoiceRecords)
             }
         }
 
-        public IEnumerable<InvoiceRecord> InvoiceRecords { get; }
+        public IEnumerable<Invoice> InvoiceRecords { get; }
```
[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/InvoiceDocument.cs:19]: Will change this, thanks

File: src/Mews.Fiscalization.Greece/Model/Invoice.cs [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public class Invoice
+    {
+        public Invoice(
+            InvoiceParty issuer,
+            InvoiceHeader invoiceHeader,
```
[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Invoice.cs:12]: Will change this, thanks

File: src/Mews.Fiscalization.Greece/Model/Invoice.cs [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+            IEnumerable<RevenueItem> revenueItems,
+            InvoiceSummary invoiceSummary,
```
[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Invoice.cs:14]: Will change this, thanks

File: src/Mews.Fiscalization.Greece/Model/Invoice.cs [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+
+        public InvoiceParty Issuer { get; }
+
+        public InvoiceParty Counterpart { get; }
+
+        public InvoiceHeader InvoiceHeader { get; set; }
+
+        public IEnumerable<Payment> PaymentMethods { get; set; }
+
+        public IEnumerable<RevenueItem> InvoiceDetails { get; }
```
[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/Invoice.cs:51]: Will change this, thanks

File: src/Mews.Fiscalization.Greece/Model/InvoiceParty.cs [hunk 13]
```diff
@@ -0,0 +1,34 @@
+﻿using Mews.Fiscalization.Greece.Model.Types;
+using System;
+
+namespace Mews.Fiscalization.Greece.Model
+{
+    public class InvoiceParty
+    {
+        private const string GreeceCountryCode = "GR";
+
+        public InvoiceParty(NotEmptyString taxNumber, CountryCode countryCode, NonNegativeInt branch = null, StringIdentifier name = null, Address invoiceRecordPartyAddress = null)
```
[the remark this answers is not in the document]
Comment (abdallahbeshi) [reply src/Mews.Fiscalization.Greece/Model/InvoiceParty.cs:10]: Makes sense, thanks.

File: src/Mews.Fiscalization.Greece/Model/Types/TaxIdentifier.cs (hunk 1 above)
Comment (abdallahbeshi) [reply src/Mews.Fiscalization.Greece/Model/Types/TaxIdentifier.cs:5]: Already discussed.

File: src/Mews.Fiscalization.Greece/Model/Invoice.cs [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+        public StringIdentifier InvoiceIdentifier { get; }
+
+        public InvoiceRegistrationNumber InvoiceRegistrationNumber { get; }
+
+        public InvoiceRegistrationNumber CanceledByInvoiceRegistrationNumber { get; }
+
+        public InvoiceParty Issuer { get; }
+
+        public InvoiceParty Counterpart { get; }
```
[the remark this answers is not in the document]
Comment (KaliCZ) [reply src/Mews.Fiscalization.Greece/Model/Invoice.cs:45]: Personally I'm okay with counterpart. But we can rename this whenever also ourselves, so it's not a priority.

File: src/Mews.Fiscalization.Greece/Model/InvoiceParty.cs (hunk 13 above)
[the remark this answers is not in the document]
Comment (KaliCZ) [reply src/Mews.Fiscalization.Greece/Model/InvoiceParty.cs:10]: There is a question: If it is from greece, does it have to be a valid `TaxIdentifier`? If yes, then we should have 1 class for `LocalInvoiceParty` (without country code and with TaxIdentifier) and 1 class for `ForeignInvoiceParty` with NonEmptyString and Country code.

[the remark this answers is not in the document]
Comment (yurypat) [reply src/Mews.Fiscalization.Greece/Model/InvoiceParty.cs:10]: Yes, it should be valid TaxIdentifier. I'll add these new classes

**Pull request merged**